staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -169,17 +174,60 @@ func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool
 	return value, nil
 }
 
+//Deprecated: GetValueFromIntOrPercent was deprecated in favor of
+//GetScaledValueFromIntOrPercent. This method was treating all int as a numeric value and all
+//strings with or without a percent symbol as a percentage value.
+func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
+	inlinedGetIntOrPercentValue := func(intOrStr *IntOrString) (int, bool, error) {
```

Comment:
I don't think this needs to be an embedded function?

Reply:
I think it's less confusing to keep the existing non-embedded function, clearly comment not to call it, and make a new one with a different name for the new function.